### PR TITLE
Fix nix build and cabal constraints

### PR DESCRIPTION
**Pull request by disassembler**
[Hydra jobset - PR #103](https://hydra.iohk.io/jobset/Cardano/cardano-addresses-pr-103#tabs-jobs)
[Hydra jobset - master](https://hydra.iohk.io/jobset/Cardano/cardano-addresses#tabs-jobs)

### CODE REVIEW

**Comment by hamishmack** [comment]
I included these changes and made some more nix fixes in #104

File: cabal.project.freeze [hunk 1]
```diff
@@ -1,2471 +0,0 @@
-constraints: any.ALUT ==2.4.0.3,
```
Comment (rvl) [inline cabal.project.freeze:1]: We need a cabal.project.freeze file - is it possible to regenerate this?

**Review by rvl — CHANGES_REQUESTED** [review]
Thanks @disassembler

**Review by rvl — APPROVED** [review]

**Comment by rvl** [comment]
I have rebased and squashed - thanks for the fixes @hamishmack @disassembler.
Unfortunately I had to give up on `cabal.project.freeze` - but `index-state` should achieve the same thing more or less.

**Pull request merged**